src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/scanneddocs/ManageScannedDocsContestedAboutToSubmitHandler.java

```diff
@@ -147,4 +159,16 @@ private void setDefaultsForDocumentTypes(UploadCaseDocumentCollection document)
 
     }
 
+    private void removeProcessedScannedDocumentsFromCase(
+        FinremCaseData caseData, List<String> processedScannedDocumentIds) {
+        processedScannedDocumentIds.forEach(id -> removeScannedDocumentFromCase(caseData, id));
+    }
+
+    private void removeScannedDocumentFromCase(FinremCaseData caseData, String id) {
+        Optional<ScannedDocumentCollection> scannedDocument = caseData.getScannedDocuments().stream()
+            .filter(sdc -> sdc.getId().equals(id))
+            .findFirst();
+
+        scannedDocument.ifPresent(sdc -> caseData.getScannedDocuments().remove(sdc));
```

Comment:
think you don't need the ifPresent, just can remove() and let the ArraList deal with nulls.

Reply:
update